modules/ingester/flush.go

```diff
@@ -229,10 +266,68 @@ func (i *Ingester) flushBlock(userID string, blockID uuid.UUID) error {
 		if err != nil {
 			return err
 		}
+
+		// Delete original wal only after successful flush
+		err = instance.ClearCompletingBlock(blockID)
+		if err != nil {
+			// Error deleting wal doesn't fail the flush
+			level.Error(log.Logger).Log("msg", "Error clearing wal", "userID", userID, "blockID", blockID.String(), "err", err)
+			metricFailedFlushes.Inc()
+		}
+
 		metricBlocksFlushed.Inc()
 	} else {
 		return fmt.Errorf("error getting block to flush")
 	}
 
 	return nil
 }
+
+func (i *Ingester) enqueue(op *flushOp, jitter bool) {
+	delay := time.Duration(0)
+
+	if jitter {
+		delay = time.Duration(rand.Float32() * float32(flushJitter))
+	}
+
+	op.at = time.Now().Add(delay)
+
+	go func() {
+		time.Sleep(delay)
+
+		// Check if shutdown initiated
+		if i.flushQueues.IsStopped() {
```

Comment:
technically still a race here which sucks. should we PR a fix upstream?

or (dare I say it) recover the panic?

Reply:
Talked to @gouthamve and the PriorityQueue is basically part of the chunks storage engine and considered maintenance mode, so first instinct was to fork it to Tempo.   That is done, now returns error instead of panic.  Leaving in the guard bool because I think it is still cleaner, to know that work is being abandoned specifically versus some other error within the queue.